### ISSUE TITLE
Insignificant roundabout exits are counted for turn instructions

### ISSUE THREAD

**Issue by martenjj** [osmandapp/OsmAnd#10028]
This is not really a routing issue, but a problem with the turn instructions counting roundabout exits when passing one which is insignificant.  Here "insignificant" mans a service road, no-access road or stub, which joins the roundabout but could not be sensibly counted as an exit and is not shown on road signs.  For illustration:

![osmand-chiswick-roundabout-issueXXXXX](https://user-images.githubusercontent.com/10631760/96117555-2e78d280-0ee2-11eb-9efa-22d31528a274.png)

Start:  51.49261 -0.28213  A406 Gunnersbury Avenue southbound, Chiswick, GB
End:  51.49096 -0.28340  A205 Chiswick High Road, Chiswick, GB
Routing profile: Internal, Car, prefer motorways, no vehicle parameters, no avoid roads
OsmAnd version: 3.8.3
Device: Samsung S10, Android 10
Map: UK England 1 Sep 2020

The route is calculated and shown on the map correctly, but the turn instructions say "Roundabout: Take 5 exit and go A205 Chiswick High Road".  However, the exits are counted to arrive at the "5th" by including the small service road on the inside of the roundabout, OSM way https://www.openstreetmap.org/way/644060010.  No driver on the ground would consider this to be an exit from the roundabout, and it is not marked on any signs.  The correct exit from the roundabout should be the 4th (1st A315 eastbound, 2nd Surrey Crescent, 3rd A4 Great West Road, 4th A205 Chiswick High Road westbound).

Unfortunately it is not possible to reliably deduce from OSM whether a road joining the roundabout is "significant" or not.  However, a useful heuristic for counting roundabout exits may be to not count "highway=service" or "highway=track", or any road tagged with "access=private" or "access=no" - *unless* the calculated route is along such a way.  It would not be a good idea to look at specialised access restrictions for the type of vehicle according to the routing profile, though - even if, for example, Surrey Crescent were tagged "hgv=no" it should still be counted as an exit because it is shown on road signs and would be seen on the ground, the directions should not change to "take the 3rd exit" just because the lorry routing profile is in use.

I have recently marked the service road access=private in OSM, and will check next month's map update to see if the same happens.  With current OSM, OSRM (Car) correctly identifies the 4th exit, but GraphHopper (Car) still considers it to be the 5th.

A similar problem can be seen at the St.Margarets roundabout, where the private service road https://www.openstreetmap.org/way/609639115 is counted as an exit.  Approaching the roundabout from the west and continuing eastwards along The Avenue says "Take the 3rd exit" although it should be the 2nd.  Also the service road https://www.openstreetmap.org/way/645061341.

**Comment by vshcherb** [comment]
Very tricky issue, I don't see any solution yet. It should be somehow specially marked with data to not allow that road for roundabout. I guess to discount all service roads will lead to other issues in different places

**Comment by aceman444** [comment]
Indeed, if the road is physically there, it is hard to discount it. You can't just say it isn't on the traffic sign and go by the sign, many signs do not show the real shape of the roundabout.

**Comment by travnick** [comment]
I have the same thing here and getting very confused, when it says to left at 7th exit, while technically there are only 3 and others are defined as
```
highway=service
service=driveway
```
Maybe to not break the whole world, then it's possible to have a switch for that?
* "Do not count `service` as a roundabout exit".

Then it's possible to have it tested before making permanent change (if it will work everywhere).

https://www.openstreetmap.org/#map=19/50.37356/19.27501

![Screenshot_20240405-121201](https://github.com/osmandapp/OsmAnd/assets/619590/e0ae9efa-7c2a-4061-9c54-41a9f9a076d7)

**Comment by DmitryAlexei** [comment]
Looks similar to https://github.com/osmandapp/OsmAnd/issues/18279
highway=service are treated as exits 
https://osmand.net/map/?start=50.374040,19.272997&end=50.374252,19.275319&profile=car#19/50.37356/19.27393

**Comment by vmicho** [comment]
I'd personally not consider driveways as exits as they are mostly thought as private access to property (just most often not marked as private as it might complicate e.g. pedestrian/bike navigation sometimes, or just not marked private as it is usually useless).
However service roads are perfectly fine to be considered as exits.

Also on the picture above, I'd announce an "exit to left", this is some special case.

**Comment by travnick** [comment]
After a while I realised, that (at least for me) anything that goes through the sidewalk is not considered as an exit.

Ignoring the driveway as exist sounds like a good idea.
